addons/website/models/res_users.py

```diff
@@ -72,10 +72,14 @@ def authenticate(cls, db, login, password, user_agent_env):
                 visitor_sudo = env['website.visitor']._get_visitor_from_request()
                 if visitor_sudo:
                     partner = env.user.partner_id
-                    partner_visitor = env['website.visitor'].sudo().search([('partner_id', '=', partner.id)])
+                    partner_visitor = env['website.visitor'].with_context(active_test=False).sudo().search([('partner_id', '=', partner.id)])
                     if partner_visitor and partner_visitor.id != visitor_sudo.id:
+                        # Link history to older Visitor and delete the newest
```

Comment:
Seems to me that the _get_visitor_from_request() should NOT create the visitor in that specific use case.
It's a little pointless to create a visitor to immediately unlink it, right?

Anyway, that would be for another task targeting master.

Reply:
No because the visitor is already created (not in authenticate process but) in dispatch process.
In authenticate, we just retrieve the visitor from the visitor's cookie.

Reply:
Ok, use case a little pourri, but ok